lib/wda/xcodebuild.js

```diff
@@ -340,6 +360,34 @@ class XcodeBuild {
     return currentStatus;
   }
 
+  async setXCTESTRUNFile () {
+    let xctestrunDeviceFileName = `${this.device.udid}_${this.platformVersion}.xctestrun`;
+
+    this.xctestrunFilePath = path.resolve(this.bootstrapPath, xctestrunDeviceFileName);
+    if (!await fs.exists(this.xctestrunFilePath)) {
+      let xctestBaseFileName = this.realDevice ? `WebDriverAgentRunner_iphoneos${this.platformVersion}-arm64.xctestrun` :
+                                `WebDriverAgentRunner_iphonesimulator${this.platformVersion}-x86_64.xctestrun`;
+      let originalXctestrunFile = path.resolve(this.bootstrapPath, xctestBaseFileName);
+      if (!await fs.exists(originalXctestrunFile)) {
+        log.errorAndThrow(`if you are using ${this.useXCTESTRUNFile} then you need to have ${originalXctestrunFile} file`);
+      }
+      await fs.copyFile(originalXctestrunFile, this.xctestrunFilePath);
```

Comment:
It's not very clear why it is done. It would be good to add a comment here about why do we need to copy the file. BTW, cannot we share the xctestrun file for multiple simulators? Isn't it read-only after build is done?

Reply:
sure I will add comments. We can't share it for one reason as we need different value of `USE_PORT` variable for different simulator.